packages/array-permutations/index.mjs

```diff
@@ -28,7 +28,7 @@ function permutations(arr) {
     for (var j = 0, len2 = partial.length; j <= len2; j++) {
       var start = partial.slice(0, j);
       var end = partial.slice(j);
-      var merged = start.concat(first, end);
+      var merged = start.concat([first], end);
```

Comment:
"first" is an item of the array being permuted. So if `arr` has type `T[]`, then:
```
arr: T[]
first: T
start: T[]
end: T[]
```
The arguments to `concat` should both be of type `T[]`.

The fact that this was (mostly) working before is due to JS's `concat` behaving somewhat like push: it can accept non-array values.  But if `T` itself is an array type, then this did not work as intended.